packages/http-client-libcurl/src/index.ts

```diff
@@ -0,0 +1,124 @@
+import { effect as T } from "@matechs/effect";
+import * as H from "@matechs/http-client";
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/pipeable";
+import * as R from "fp-ts/lib/Record";
+import * as C from "node-libcurl";
+import path from "path";
+
+const certfile = path.join(__dirname, "../cacert-2019-11-27.pem");
+
+export const libcurl: H.Http = {
+  http: {
+    request: <I, E, O>(
+      method: H.Method,
+      url: string,
+      headers: Record<string, string>,
+      body: I
+    ): T.Effect<T.NoEnv, H.HttpError<E>, H.Response<O>> =>
+      T.async(done => {
+        const req = new C.Curl();
+        const reqHead = [
+          "Content-Type: application/json",
+          ...pipe(
+            headers,
+            R.collect((k, v) => `${k}: ${v}`)
+          )
+        ];
+
+        req.setOpt("URL", url);
+        req.setOpt("CAINFO", certfile);
```

Comment:
Pondering if should not be a dep.. :)

Reply:
this is to be though there are pro and cons of shipping the bundle I think we should allow consumer to overwrite it but we should also ship it given libcurl doesn't integrate well with systems

Reply:
made at least overridable